src/mods/visual/visual.ts

```diff
@@ -60,7 +64,9 @@ const rectSchema = struct({
 export type PolyStyle = Partial<TypeOf<typeof polySchema>['s']>;
 const polySchema = struct({
 	...variant('p'),
-	points: vector(array(2, 'double')),
+	// to allow room names for map visuals the points have to be saved in a less optimal way as objects.
```

Comment:
I'm not familiar enough with map visuals to know this, but would it not be possible to parse the room name as soon as the player gives it to us and store the result as raw coordinates? Basically when you add properties to a schema it increases the size of the data structure, even in the case of `optional`. Since these payloads are often pretty huge (around 500kb) I definitely don't want them to be any bigger than they have to be.

Reply:
For the poly style there are two ways the coordinates can be send to the user ... as a simple array of coordinates like [[x,y],[x,y]...] or as room-positions [{x,y,n},{x,y,b},...]. I guess the second form was implemented when map visuals where added.
I now changed the database implementation to always use the second form because it works for room and for map visuals. 
I guess there is no easy way to allow both styles saved to the database? Maybe it makes sense to store it a different way and change the structure during the json serialization when it gets send over the socket?

Reply:
I took a quick peek at engine's implementation.. I see that they're storing `{ roomName, x, y}` but what we can do is something like this:
```js
const { rx, ry } = parseRoomName(roomName);
return { x: rx + x / 50, y: ry + y };
```

Then on the way to the client we can convert it into the format expected.

Reply:
I have implemented the room storage in doubles for all the visual types... so no need to change any schemas